### PR TITLE
DAITDS-142 Application support

### PR DESCRIPTION
**Pull request by pcatalini**

### CODE REVIEW

File: ns1/resource_application.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	}
+	reqConf := defaultConfig["request_timeout_millis"]
+	if reqConf != nil {
+		reqInt, _ := strconv.Atoi(reqConf.(string))
+		d.RequestTimeoutMillis = reqInt
+	}
+	return
+}
+
+// ApplicationCreate creates the given zone in ns1
```
Comment (Zach-Johnson) [inline ns1/resource_application.go:147]: The comment on this is incorrect I think

File: ns1/resource_application.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	if resp, err := client.Applications.Create(app); err != nil {
+		return ConvertToNs1Error(resp, err)
+	}
+	if err := resourceApplicationToResourceData(d, app); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ApplicationRead reads the given zone data from ns1
```
Comment (Zach-Johnson) [inline ns1/resource_application.go:161]: Comment looks wrong

File: ns1/resource_application.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+func ApplicationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ns1.Client)
+	app, resp, err := client.Applications.Get(d.Id())
+	if err != nil {
+		if err == ns1.ErrApplicationMissing {
```
Comment (Zach-Johnson) [inline ns1/resource_application.go:166]:
``` suggestion
		if errors.Is(err, ns1.ErrApplicationMissing) {
```

File: ns1/resource_application.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+		return ConvertToNs1Error(resp, err)
+	}
+
+	if err := resourceApplicationToResourceData(d, app); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ApplicationDelete deletes the given zone from ns1
```
Comment (Zach-Johnson) [inline ns1/resource_application.go:180]: fix comment

File: ns1/resource_application.go [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+func ApplicationDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ns1.Client)
+	resp, err := client.Applications.Delete(d.Id())
+	d.SetId("")
+	return ConvertToNs1Error(resp, err)
+}
+
+// ApplicationUpdate updates the zone with given params in ns1
```
Comment (Zach-Johnson) [inline ns1/resource_application.go:188]: fix comment

File: ns1/resource_application_test.go [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+				Check:  testAccCheckApplicationExists("ns1_application.it", &application),
+			},
+		},
+	})
+}
+
+func testAccCheckApplicationDefaultConfig(app *pulsar.Application, expected pulsar.DefaultConfig) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if app.DefaultConfig != expected {
+			return fmt.Errorf("zone: got: %v want: %v", app.JobsPerTransaction, expected)
```
Comment (Zach-Johnson) [inline ns1/resource_application_test.go:136]: don't think this should say zone, various other spots with this as well

File: website/docs/r/application.html.markdown [hunk 7]
```diff
@@ -0,0 +1,82 @@
+---
+layout: "ns1"
+page_title: "NS1: ns1_application"
+sidebar_current: "docs-ns1-resource-application"
+description: |- Provides a NS1 Pulsar Application resource.
+---
+
+# ns1\_application
```
Comment (Zach-Johnson) [inline website/docs/r/application.html.markdown:8]: is this supposed to have the `\` in it?

**Review by Zach-Johnson — CHANGES_REQUESTED** [review]
Looks good, just a few minor things noted

File: website/docs/r/application.html.markdown [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  0, the primary NSONE Global Network. Normally, you should not have to worry about this.
+* `use_xhr` - (Computed) - Whether to use XMLHttpRequest (XHR) when taking measurements.
+* `static_values` - (Computed) - Indicates whether or not to skip aggregation for this job's measurements
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `dns_servers` - (Computed) Authoritative Name Servers.
+* `hostmaster` - (Computed) The SOA Hostmaster.
```
Comment (mvkushnir) [inline website/docs/r/application.html.markdown:70]: What are these?

Comment (pcatalini) [reply website/docs/r/application.html.markdown]: i forget to remove, thanks good catch

File: website/docs/r/application.html.markdown [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+#### Default config
+
+* `http` - (Required) Indicates whether or not to use HTTP in measurements.
+* `https` - (Optional) Indicates whether or not to use HTTPS in measurements.
+* `request_timeout_millis` - (Optional) Maximum timeout per request.
+* `job_timeout_millis` - (Computed) - Maximum timeout per job
+  0, the primary NSONE Global Network. Normally, you should not have to worry about this.
+* `use_xhr` - (Computed) - Whether to use XMLHttpRequest (XHR) when taking measurements.
+* `static_values` - (Computed) - Indicates whether or not to skip aggregation for this job's measurements
```
Comment (mvkushnir) [inline website/docs/r/application.html.markdown:63]: These should be optional, not computed (as they are up above)

**Review by mvkushnir — APPROVED** [review]
Looks good, just one final change

**Review by Zach-Johnson — APPROVED** [review]
lgtm!

**Pull request merged**